app/views/public/loans/show.html.slim

```diff
@@ -13,27 +13,32 @@
       .detail_wrap
         h2.section_title = @loan.display_name
 
-        / Carousel
-        - if @pictures
-          #myCarousel.carousel.slide
-            - if @pictures.count > 1
-              ol.carousel-indicators
-                - for i in (0..(@pictures.count-1)) do
-                  li data-slide-to=i data-target="#myCarousel"
-            / Carousel items
-            a href=public_gallery_path
-              .carousel-inner
-                - @pictures.each do |picture|
-                  .item.picture style=("background-image: url(#{picture.item.large.url});")
-            / Carousel nav
-            - if @pictures.count > 1
-              a.carousel-control.left data-slide="prev" href="#myCarousel"  &lsaquo;
-              a.carousel-control.right data-slide="next" href="#myCarousel"  &rsaquo;
-
         / .lend
         /   a.button.lend_button.pull-right href="#"
         / h5.coop_name Current Report
 
+        a href=public_gallery_path
+          #loanImageCarousel.carousel.slide data-ride="carousel"
+            ol.carousel-indicators
+              - remainingPictures = @pictures.length
```

Comment:
@meltheadorable There is likely a cleaner way to do this loop. Please let me know how I can improve it. Thank you!

Reply:
I'm not fully grasping the goal of this loop, what's it doing?

Reply:
Without being 100% sure what the goal is, this might be what you're looking for:

```slim
ol.carousel-indicators
  - @pictures.each.with_index do |picture, i|
    li data-target="#loanImageCarousel" data-slide-to="#{i}" class="#{'active' if i.zero?}"
```